src/na/na_ucx.c

```diff
@@ -1695,12 +1713,16 @@ na_ucp_listener_conn_cb(ucp_conn_request_h conn_request, void *arg)
         error, "conn attributes contain no client addr");
 
     /* Lookup address from table */
-    addr_key = (ucs_sock_addr_t){
+    addr_key = (ucs_sock_addr_t) {
         .addr = (const struct sockaddr *) &conn_request_attrs.client_address,
         .addrlen = sizeof(conn_request_attrs.client_address)};
     na_ucx_addr = na_ucx_addr_map_lookup(&na_ucx_class->addr_map, &addr_key);
-    NA_CHECK_SUBSYS_ERROR_NORET(addr, na_ucx_addr != NULL, error,
-        "An entry is already present for this address");
+
+    if (na_ucx_addr != NULL) {
+        NA_LOG_SUBSYS_WARNING(
+            addr, "An entry is already present for this address");
+        na_ucx_addr_release(na_ucx_addr, true, NULL);
```

Comment:
I think I asked on the wrong PR yesterday, Joe answered me but I still don't quite understand. 
Is it ok to release the address without checking refcount? What if this address and associated EP are still using by someone else, will it cause memory corruption or other issue?

Reply:
it should be ok, we don't have much other choice I think, this fix was made by Joe for another issue and I just brought it up as part of this PR at the same time. We could add an extra warning there for sanity checks. If we end up in this code path this means that for this specific source address a new endpoint is established by UCX since this is triggered by the listener and afaik that means that the previous endpoint is no longer usable. Release will then force close the previous EP. I also don't see anything there that would cause a corruption since memory is not released.

Reply:
After spending some more time on it, I think the wise thing to do in this case would be to defer the connection request and let it complete (calling accept on it) only once the previous address has been fully released. As I said I'll remove that code from that PR and make a separate PR since this is unrelated to the original issue.

Reply:
Reworked this bit in https://github.com/mercury-hpc/mercury/pull/821